### ISSUE TITLE
Rework default player configuration to support parallel runs

### ISSUE THREAD

**Issue by Benjamin-Etheredge** [hsahovic/poke-env#208]
The current default player configuration generator will cause errors when two runs are kicked off. This is likely due to collisions with the default configuration counter.

**Comment by hsahovic** [comment]
Hey @Benjamin-Etheredge,

That sounds like a very good catch. Do you have a code snippet I could use to reproduce it?

**Comment by Benjamin-Etheredge** [comment]
I have a fix for it. I'm trying to get it pushed to a new branch now.

**Comment by Benjamin-Etheredge** [comment]
I don't have a code snippet to reproduce it. I noticed it when trying to run the same code twice in two shells.

**Comment by Benjamin-Etheredge** [comment]
There may also be an issue with passing specific player configurations through extended PlayerEnvs. I'm still investigating that one as it could be an issue with my code. It was easier to just rewrite the default configuration generator rather than investigate why my custom player names weren't working. The rework now works for parallel runs though.

**Comment by hsahovic** [comment]
I don't think that `uuid`s is the way to go - using the counter-based solution creates a clearer sequence of player usernames, which can be useful when debugging or running certain types of algorithms (eg. genetic algorithms with a clear number / generation correlation).
If you can provide a snippet to reproduce this issue, I'll gladly issue a fix (probably using a `Lock` object).

**Comment by Benjamin-Etheredge** [comment]
I'm not sold on uuid either. It is ugly and hard to track. I had to start logging player names to follow what was going on.
I like the more sequential nature, but it leads to collisions when two default configurations are used. 

I reproduced it by making a python file with the example code (heads up, there is a typo in the indentions of that example code too)
```
# -*- coding: utf-8 -*-
import asyncio
import time

from poke_env.player.player import Player
from poke_env.player.random_player import RandomPlayer


class MaxDamagePlayer(Player):
    def choose_move(self, battle):
        # If the player can attack, it will
        if battle.available_moves:
            # Finds the best move among available ones
            best_move = max(battle.available_moves, key=lambda move: move.base_power)
            return self.create_order(best_move)

        # If no attack is available, a random switch will be made
        else:
            return self.choose_random_move(battle)


async def main():
    start = time.time()

    # We create two players.
    random_player = RandomPlayer(
        battle_format="gen8randombattle",
    )
    max_damage_player = MaxDamagePlayer(
        battle_format="gen8randombattle",
    )

    # Now, let's evaluate our player
    await max_damage_player.battle_against(random_player, n_battles=1000)

    print(
        "Max damage player won %d / 100 battles [this took %f seconds]"
        % (
            max_damage_player.n_won_battles, time.time() - start
        )
    )


if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(main())
 ```
 
 1. Make a python file with that.
 2. Open two shells.
 3. Start the code in one shell.
 4. Start the code again in another shell.
 The second shell will error out with something like 'Someone is already using that name'.

**Comment by Benjamin-Etheredge** [comment]
Is there a way to store player names as a side effect? Or maybe to catch the "Someone is already using that name" in the default configuration and keep incrementing till the name is free?

**Comment by Benjamin-Etheredge** [comment]
An additional note for why this needs updating: 
I'm exploring using openai baselines (and other forks). One key element for algorithms is parallel environments. I'm able to get parallel environments working more easily thanks to patching default configuration generation to not collide on player names.

**Comment by cameronangliss** [comment]
I believe the fix for this is implemented in #731, where you can now do `AccountConfiguration.generate(<KEY>, rand=True)`, and a random sequence of 5 digits is appended to the end of the username. This is used now in `PokeEnv` to generate the usernames for the two EnvPlayer's that play against each other for exactly the reason described in this issue, that it is useful to have multiple parallel runs be able to run without interfering with each other in the same showdown server.

Closing now.
